timetomodel/speccing.py

```diff
@@ -305,7 +305,8 @@ def reverse_sum(resampler, **kwargs):
         # Convert to PeriodIndex for desired behavior under resampling (specifically, binning the right-most period)
         # remember timezone before resampling, see https://github.com/pandas-dev/pandas/issues/28039
         tz = data.index.tzinfo
-        data.index = data.index.to_period(event_resolution)
+        if event_resolution != timedelta(hours=0):
```

Comment:
Can you add a reason why we do this for resampling, and why we don't need it for zero-resolution?

Reply:
I extended the inline comments accordingly.